### PR TITLE
Feat(eos_cli_config_gen): Support accounting logging

### LINKED ISSUE #3045: [eos_cli_config_gen] aaa_accounting.exec.[console, default] missing boolean logging
### Enhancement summary

I would like to see support in eos_cli_config_gen for below commands

```
aaa accounting exec console start-stop logging
aaa accounting exec default start-stop logging
```

Ansible validation error

`
TASK [arista.avd.eos_cli_config_gen : Generate eos intended configuration] **********************************************************************************************************
ERROR! [dc3-leaf1]: 'Validation Error: aaa_accounting.exec.console': Unexpected key(s) 'logging' found in dict.
ERROR! [dc3-leaf1]: 'Validation Error: aaa_accounting.exec.default': Unexpected key(s) 'logging' found in dict.
`

```
aaa_accounting:
  exec:
    console:
      type: start-stop
      logging: true # Missing in data model but possible to configure in EOS
    default:
      type: start-stop
      logging: true # Missing in data model but possible to configure in EOS
  commands:
    console:
      - commands: all
        type: start-stop
        logging: true # Implemented in data model and schema
    default:
      - commands: all
        type: start-stop
        logging: true # Implemented in data model and schema
```

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Being able to use eos_cli_config_gen to generate aaa accounting exec command and using syslog as the logging destination instead of tacacs+ or radius

### Describe the solution you would like

Same behavior as aaa_accounting.commands[console, default] currently have, with support for boolean `logging`

### Describe alternatives you have considered

_No response_

### Additional context

Test ran on AVD 4.1.0

Docs used https://avd.sh/en/stable/roles/eos_cli_config_gen/docs/input-variables.html#aaa-accounting

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by colinmacgiolla**
## Change Summary

We don't currently have to option to enable (or not logging) for AAA accounting exec e.g.
```
aaa accounting exec console start-stop logging
aaa accounting exec default start-stop logging
```
though looking into this I can see 2 other problems with the model that would be breaking;
1. We assume only a single group
2. There is no requirement for a group, but we won't render either line unless a group is defined

```
0ac0f6ee6528(config)#aaa accounting exec console start-stop logging ?
  group    Specify server group
  logging  Use syslog for accounting
  <cr>
```
This changes fixes #3045, and the second issue, but leaves the the first alone as that would be a breaking change

## Related Issue(s)

Fixes #3045 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Add a boolean for logging, similar to how is done elsewhere, and update the schemas to reflect this
```yml
aaa_accounting:
  exec:
    console:
      type: start-stop
      logging: true 
    default:
      type: start-stop
      logging: true 
  commands:
    console:
      - commands: all
        type: start-stop
        logging: true
    default:
      - commands: all
        type: start-stop
        logging: true
```

## How to test

Molecule tests updated 

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-accounting.j2 [hunk 1]
```diff
@@ -1,7 +1,20 @@
 {# eos - aaa accounting #}
 {% if aaa_accounting is arista.avd.defined %}
-{%     if aaa_accounting.exec.console.type is arista.avd.defined and aaa_accounting.exec.console.group is arista.avd.defined %}
-aaa accounting exec console {{ aaa_accounting.exec.console.type }} group {{ aaa_accounting.exec.console.group }}
+{%     if aaa_accounting.exec.console.type is arista.avd.defined %}
+{%         set aaa_accounting_exec_console_cli = "aaa accounting exec console" %}
+{%         if aaa_accounting.exec.console.type is arista.avd.defined('none') %}
+{%             set aaa_accounting_exec_console_cli = aaa_accounting_exec_console_cli ~ " none" %}
+{{ aaa_accounting_exec_console_cli }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-accounting.j2]:
```suggestion
```
You already print this in the end - outside the if/else block.
We should test this condition, since that would exhibit this issue.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-accounting.j2 [hunk 2]
```diff
@@ -17,8 +30,21 @@ aaa accounting exec console {{ aaa_accounting.exec.console.type }} group {{ aaa_
 {{ aaa_accounting_commands_commands_console_cli }}
 {%         endfor %}
 {%     endif %}
-{%     if aaa_accounting.exec.default.type is arista.avd.defined and aaa_accounting.exec.default.group is arista.avd.defined %}
-aaa accounting exec default {{ aaa_accounting.exec.default.type }} group {{ aaa_accounting.exec.default.group }}
+{%     if aaa_accounting.exec.default.type is arista.avd.defined %}
+{%         set aaa_accounting_exec_default_cli = "aaa accounting exec default" %}
+{%         if aaa_accounting.exec.default.type is arista.avd.defined('none') %}
+{%             set aaa_accounting_exec_default_cli = aaa_accounting_exec_default_cli ~ " none" %}
+{{ aaa_accounting_exec_default_cli }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-accounting.j2]:
```suggestion
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/aaa.yml
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/aaa.yml:1]: Please add a test for the `none` case. Maybe add another host `aaa-2` with some other combinations. (only include the key you wish to test. Don't add radius_servers etc).
Logging should be tested under the default as well, since it is two seperate bits of code.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
Thank you for fixing this.
Please also update the documentation template, where the table has a "logging" column, but we are not populating it.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/aaa.yml
Comment (colinmacgiolla) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/aaa.yml:1]: I was wondering about that but thought to leave `logging` out of default (since its basically the same code) as a test for not rendering logging when its not configured... Better to add it?

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/aaa.yml [hunk 3]
```diff
@@ -91,6 +91,7 @@ aaa_accounting:
     console:
       type: start-stop
       group: TACACS
+      logging: true
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/aaa.yml:94]: ~Let's put logging on console in one of the files and on "default" in the other file. Then we test all variants. Thanks.~ nvm, with "none" it is ignored. So please just add it under both in this file.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/aaa-accounting.j2 [hunk 4]
```diff
@@ -12,8 +12,10 @@
 
 | Type | Commands | Record type | Group | Logging |
 | ---- | -------- | ----------- | ----- | ------- |
-{%     if aaa_accounting.exec.console.type is arista.avd.defined and aaa_accounting.exec.console.group is arista.avd.defined %}
-| Exec - Console | - | {{ aaa_accounting.exec.console.type }} | {{ aaa_accounting.exec.console.group }} | - |
+{%     if aaa_accounting.exec.console.type is arista.avd.defined  %}
+{%         set logging = aaa_accounting.exec.console.logging | arista.avd.default("False") %}
+{%         set group = aaa_accounting.exec.console.group | arista.avd.default(" - ") %}
+| Exec - Console | - | {{ aaa_accounting.exec.console.type }} | {{ group }} | {{ logging }} |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/aaa-accounting.j2]: Aligning to our regular implementation and avoid overwriting common names outside of a loop.
```suggestion
{%         set aaa_accounting_logging = aaa_accounting.exec.console.logging | arista.avd.default("-") %}
{%         set aaa_accounting_group = aaa_accounting.exec.console.group | arista.avd.default("-") %}
| Exec - Console | - | {{ aaa_accounting.exec.console.type }} | {{ aaa_accounting_group }} | {{ aaa_accounting_logging }} |
```
and please do the same below.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/aaa.yml
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/aaa.yml:1]: Yeah lets test it on both code pieces. I know we won't test the "unset" variant, but that is fine for now.

**Review by ClausHolbechArista — APPROVED** [review]
